code/datums/components/proximity_monitor.dm

```diff
@@ -286,3 +286,162 @@
 	. = ..()
 	if(active && AM != monitor.hasprox_receiver && !(AM in monitor.nested_receiver_locs))
 		monitor.hasprox_receiver.HasProximity(AM)
+
+/// A custom proximity monitor used for tracking players around a table of cards.
+
+/datum/component/proximity_monitor/table
+	/// How far away you can be (in terms of table squares).
+	var/max_table_distance
+	/// How far away you can be (euclidean distance).
+	var/max_total_distance
+	/// The UID of the deck
+	var/deck_uid
+	/// Whether the monitors created should be visible. Used for debugging.
+	var/monitors_visible = FALSE
+
+/datum/component/proximity_monitor/table/Initialize(_radius = 1, _always_active = FALSE, _max_table_distance = 5)
+	max_table_distance = _max_table_distance
+	max_total_distance = _max_table_distance
+	. = ..(_radius, _always_active)
+	if(istype(parent, /obj/item/deck))
+		// this is important for tracking traits and attacking multiple cards. so it's not a true UID, sue me
+		var/obj/item/deck/D = parent
+		deck_uid = D.main_deck_id
+	else
+		deck_uid = parent.UID()
+	refresh()
+
+/datum/component/proximity_monitor/table/proc/refresh()
+	var/list/tables = list()
+	var/list/prox_mon_spots = list()
+	crawl_along(get_turf(parent), tables, prox_mon_spots, 0)
+	QDEL_LIST_CONTENTS(proximity_checkers)
+	create_prox_checkers()
+
+/// Crawl along an extended table, and return a list of all turfs that we should start tracking.
+/datum/component/proximity_monitor/table/proc/crawl_along(turf/current_turf, list/visited_tables = list(), list/prox_mon_spots = list(), distance_from_start)
+	var/obj/structure/current_table = locate(/obj/structure/table) in current_turf
+
+	if(QDELETED(current_table))
+		// if there's no table here, we're still adjacent to a table, so this is a spot you could play from
+		prox_mon_spots |= current_turf
+		return
+
+
+	if(current_table in visited_tables)
+		return
+
+	visited_tables |= current_table
+	prox_mon_spots |= current_turf
+
+
+	if(distance_from_start + 1 > max_table_distance)
+		return
+
+	for(var/direction in GLOB.alldirs)
+		var/turf/next_turf = get_step(current_table, direction)
+		if(get_dist_euclidian(get_turf(parent), next_turf) > max_total_distance)
+			continue
+		.(next_turf, visited_tables, prox_mon_spots, distance_from_start + 1)
+
+/datum/component/proximity_monitor/table/create_prox_checkers()
+	update_prox_checkers(FALSE)
+
+/datum/component/proximity_monitor/table/proc/update_prox_checkers(clear_existing = TRUE)
```

Comment:
Should we add an autodoc to this to explain what is going on here?

Reply:
Yeah, I can put some better documentation in here.